components/org.wso2.carbon.event.outbound.adapter.websubhub/src/main/java/org/wso2/carbon/event/outbound/adapter/websubhub/util/WebSubHubEventAdapterUtil.java

```diff
@@ -0,0 +1,324 @@
+     *
+     * @param securityEventTokenPayload Security Event Token object.
+     * @param tenantDomain              Tenant domain.
+     * @param topic                     Topic name.
+     * @param webSubHubBaseUrl          Web sub hub base url.
+     * @throws WebSubAdapterException on error while publishing the events.
+     */
+    public static void makeAsyncAPICall(SecurityEventTokenPayload securityEventTokenPayload, String tenantDomain,
+                                        String topic, String webSubHubBaseUrl) throws WebSubAdapterException {
+
+        String url = buildURL(topic, tenantDomain, webSubHubBaseUrl, PUBLISH);
+
+        HttpPost request = new HttpPost(url);
+        request.setHeader(ACCEPT, TYPE_APPLICATION_JSON);
+        request.setHeader(CONTENT_TYPE, TYPE_APPLICATION_JSON);
+        request.setHeader(CORRELATION_ID_REQUEST_HEADER, getCorrelationID());
+
+        ObjectMapper mapper = new ObjectMapper();
+        String jsonString;
+        try {
+            jsonString = mapper.writeValueAsString(securityEventTokenPayload);
+            request.setEntity(new StringEntity(jsonString));
+        } catch (IOException e) {
+            throw handleClientException(ERROR_PUBLISHING_EVENT_INVALID_PAYLOAD);
+        }
+
+        CloseableHttpAsyncClient client;
+        try {
+            client = WebSubHubEventAdapterDataHolder.getInstance().getClientManager()
+                    .getClient(tenantDomain);
+        } catch (IOException e) {
+            throw handleServerException(ERROR_GETTING_ASYNC_CLIENT, e, tenantDomain);
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Publishing event data to WebSubHub. URL: " + url + " tenant domain: "
+                    + tenantDomain);
+        }
+
+        client.execute(request, new FutureCallback<HttpResponse>() {
+            @Override
+            public void completed(final HttpResponse response) {
+
+                int responseCode = response.getStatusLine().getStatusCode();
+                if (log.isDebugEnabled()) {
+                    log.debug("WebSubHub request completed. Response code: " + responseCode);
+                }
+                if (responseCode == 200 || responseCode == 201 || responseCode == 202 || responseCode == 204) {
+                    // Check for 200 success code range.
+                    if (log.isDebugEnabled()) {
+                        String jsonString;
+                        try {
+                            jsonString = EntityUtils.toString(response.getEntity());
+                            JSONParser parser = new JSONParser();
+                            JSONObject json = (JSONObject) parser.parse(jsonString);
+                            log.debug("Response data: " + json);
+                        } catch (IOException | ParseException e) {
+                            log.error("Error while reading WebSubHub event publisher response. ", e);
```

Comment:
make it log.debug

Reply:
resolve in https://github.com/wso2-extensions/identity-outbound-adapters/pull/1/commits/b3a5860dc1035b7d44a6724b05589567293df9b0